src/task.c

```diff
@@ -1259,6 +1259,7 @@ task_handler (gpointer user_data)
               stop_uploader (app_data);
 
           rstrnt_upload_logs (task, app_data, soup_session, app_data->cancellable);
+          rstrnt_close_logs (task);
```

Comment:
I'm asking questions to make sure I understand.  Because g_hash_table_remove was NOT called, rstrnt_task_log_data_destroy() was never executed.  Is that correct?

Reply:
Yes, that's correct. Prior to the new call to `rstrnt_close_logs` logs were only added to the hash table, not removed. A debugger session on `restraintd` confirmed that log files were closed as a result of the `g_hash_table_remove` call.